### DISCUSSION TITLE
Required Workflows break repository auto_init

### DISCUSSION THREAD

**Discussion by ngoduykhanh** [community/community#58854] [Repositories]
### Select Topic Area

Product Feedback

### Body

When creating a new GitHub repository either via UI or API, we have the option to initialize the repo with a `README.md` file. This works perfectly fine until we introduce the Required Workflows in the organization. If the Required Workflows is enabled for all repositories, the newly created repository won't have any content, it will just be an empty repository. If you try to clone and push the first commit ever to the default (`main`) branch, you will receive the following error:

> Organization mandated required workflow checks need to run successfully to be able merge to this branch. Please push changes to a new branch and create a pull request to run the checks.

According to GitHub's blog post https://github.blog/changelog/2023-03-10-github-actions-required-workflows-improvements/.

> Blocking direct push: Direct pushes are now blocked on branches of the repositories where required workflows are enforced. To push to a branch where required workflows are enforced at the organizational level, create a pull request to make the necessary changes. If you want to allow direct pushes for a particular repository, you must remove the repository as a target from respective required workflows.

This makes sense only for the repositories that already have the default branch pushed. For a new, empty repository, it shouldn't prevent us from initializing the default branch for the first time.

WDYT? I know that Required Workflows is still under Beta, any enhancement will be made soon?

**Comment by Tarunvaka** [comment]
You need to create a new repo

**Reply by ngoduykhanh** [reply]
Sorry, I don't get your answer. This issue happens to the new repo creation.

**Comment by roland-at-handshake** [comment]
I'm seeing this same issue. Unless I'm misunderstanding, this effectively prevents us from creating any new repos without temporarily disabling the required workflow.

**Comment by ngoduykhanh** [comment]
It seems like Repository Rulset will replace the Required workflows
https://github.blog/changelog/2023-08-02-github-actions-required-workflows-will-move-to-repository-rules/

**Comment by laura-maree-sp** [comment]
Was there any update to this after the move to repository rulesets? I'm having the same issue where a required workflow configured through a repository ruleset is breaking the auto_init of a new repository.
